src/app/Fake.Tools.SignTool/SignTool.fs

```diff
@@ -0,0 +1,403 @@
+    let filesList = files |> List.ofSeq
+    let getTimeout = Option.defaultValue (TimeSpan.FromSeconds (10.0 * float (List.length filesList)))
+    let getToolPath = function
+        | Some p -> p
+        | None ->
+            match signtoolexeLocator () with
+            | Some p -> p
+            | None -> failwith "SignTool failed: Could not locate signtool.exe. Make sure you have Windows SDKs installed or provide direct path in the ToolPath option."
+    let getWorkingDir = Option.defaultValue (Directory.GetCurrentDirectory())
+    let signtoolPath, signtoolTimeout, signtoolWorkingDir =
+        match toolOptions with
+        | Some o -> getToolPath o.ToolPath, getTimeout o.Timeout, getWorkingDir o.WorkingDir
+        | None -> getToolPath None, getTimeout None, getWorkingDir None
+    // if there are any options, join them with a space and prepend a space separator, otherwise nothing
+    let optionsString = String.Join(" ", options) |> fun o -> if String.isNullOrWhiteSpace o then String.Empty else (" " + o)
+    // if there are any files, quote them and join them with a space and prepend a space separator, otherwise nothing
+    let filesString = if List.isEmpty filesList then String.Empty else (" \"" + String.Join("\" \"", filesList) + "\"")
+    let signtoolArgs = command + optionsString + filesString
+
+    runner signtoolPath signtoolArgs signtoolWorkingDir signtoolTimeout
+
+
+/// default runner
+let internal defaultRunner (signtoolPath: string) (signtoolArgs: string) (signtoolWorkingDir: string) (signtoolTimeout: TimeSpan) =
+    let stdOut = StringBuilder()
+    let stdErr = StringBuilder()
+    let result =
+        CreateProcess.fromRawCommandLine signtoolPath signtoolArgs
+        |> CreateProcess.withWorkingDirectory signtoolWorkingDir
+        |> CreateProcess.withTimeout signtoolTimeout
+        |> CreateProcess.redirectOutput
+        |> CreateProcess.withOutputEvents (stdOut.AppendLine >> ignore) (stdErr.AppendLine >> ignore)
+        |> Proc.run
+    Trace.log (stdOut.ToString())
+    if result.ExitCode <> 0 then
+        sprintf "SignTool failed: %s" (stdErr.ToString())
+        |> TraceSecrets.guardMessage
+        |> failwith
+
+/// default signtool.exe locator
+let internal defaultSigntoolexeLocator () =
+    let winSdksDirs = [
+        @"[ProgramFilesX86]\Windows Kits\8.1\bin\x86"
+        @"[ProgramFilesX86]\Windows Kits\10\bin\**\x86"
+    ]
+    ProcessUtils.tryFindFile winSdksDirs "signtool.exe"
+
+
+let private timestamping timestamp = seq {
+    match timestamp with
+    | Some t ->
+        let serverUrl = t.ServerUrl |> Option.defaultValue "http://timestamp.digicert.com"
+        yield match t.Algorithm with
+              | Some SHA1 -> sprintf "/t \"%s\"" serverUrl
+              | Some SHA256 | None -> sprintf "/tr \"%s\" /td \"sha256\"" serverUrl
```

Comment:
I'd prefer to use the Arguments API as (briefly) documented in https://fake.build/core-process.html and https://fake.build/apidocs/v5/fake-core-argumentsmodule.html
It handles proper escaping and edge cases accordingly

Reply:
I'll look into it.